subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -192,6 +241,9 @@ uint8_t ll_adv_params_set(uint16_t interval, uint8_t adv_type,
 	}
 
 	adv->is_created = 1;
+#if defined(CONFIG_BT_CTLR_HCI_ADV_HANDLE_MAPPING)
+	adv->hci_handle = handle;
```

Comment:
this won't work since currently `handle` passed here is still a ULL handle, not HCI handle.
I think easiest way to fix this would be to populate `hci_handle` in `ll_adv_set_by_hci_handle_get_or_new` since it will not be used anyway until `is_created` is set so will update.